plugins/save-ansi.py

```diff
@@ -0,0 +1,165 @@
+
+    fg_code = get_color_code(fg)
+    codes.append(f'38;5;{fg_code}')
+
+    if bg:
+        bg_code = get_color_code(bg)
+        codes.append(f'48;5;{bg_code}')
+
+    return '\033[' + ';'.join(codes) + 'm' if codes else ''
+
+def export_frame(rows, x_col, y_col, text_col, color_col):
+    """Export a single frame to ANSI text."""
+    if not rows:
+        return ''
+
+    # Find max_x and max_y
+    max_x = 0
+    max_y = 0
+    for row in rows:
+        x = int(x_col.getValue(row))
+        y = int(y_col.getValue(row))
+        text = str(text_col.getValue(row))
+        max_x = max(max_x, x + len(text) - 1)
+        max_y = max(max_y, y)
+
+    # Create grid
+    grid = [[None] * (max_x + 1) for _ in range(max_y + 1)]
+
+    # Place characters
+    for row in rows:
+        x = int(x_col.getValue(row))
+        y = int(y_col.getValue(row))
+        text = str(text_col.getValue(row))
+        color_str = str(color_col.getValue(row))
+        attributes, fg, bg = parse_color(color_str)
+        for i, char in enumerate(text):
+            grid[y][x + i] = (char, attributes, fg, bg)
+
+    # Generate output
+    output = ''
+    for y in range(max_y + 1):
+        line = ''
+        prev_codes = None
+        for x in range(max_x + 1):
+            cell = grid[y][x]
+            if cell:
+                char, attributes, fg, bg = cell
+                codes = get_escape_codes(attributes, fg, bg)
+                if codes != prev_codes:
+                    line += codes
+                    prev_codes = codes
+                line += char
+            else:
+                if prev_codes != '\033[0m':
+                    line += '\033[0m'
+                    prev_codes = '\033[0m'
+                line += ' '
+        line += '\033[0m'
+        output += line + '\n'
+    return output
+
+def save_ansi(self):
+    """Save the current sheet as an ANSI text file, exporting only rows with empty frame and type."""
+    required_columns = ['x', 'y', 'text', 'color']
+    columns_dict = {col.name: col for col in self.columns}
+    for col_name in required_columns:
+        if col_name not in columns_dict:
+            vd.fail(f'Missing column: {col_name}')
+
+    x_col = columns_dict['x']
+    y_col = columns_dict['y']
+    text_col = columns_dict['text']
+    color_col = columns_dict['color']
+    frame_col = columns_dict.get('frame', None)
+    type_col = columns_dict.get('type', None)
+
+    # Filter rows where both frame and type are empty
+    filtered_rows = []
+    for row in self.rows:
+        frame_value = frame_col.getValue(row) if frame_col else None
+        type_value = type_col.getValue(row) if type_col else None
+        # Consider a value empty if it is None or an empty string
+        if (frame_value is None or frame_value == '') and (type_value is None or type_value == ''):
+            filtered_rows.append(row)
+
+    if not filtered_rows:
+        vd.warning('Drawing is animation; cannot export as ANSI.')
```

Comment:
In general, use `fail` instead of `warning` + `return`. It's a minor difference but it causes the command to abort (via exception) instead of complete, which has some other consequences (like not being saved to the cmdlog, for instance).